src/components/PostsList.tsx

```diff
@@ -1,85 +1,76 @@
+            <th>Title</th>
+            <th> </th>
+          </tr>
+        </thead>
 
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link is-light"
+        <tbody>
+          {posts.map(post => (
+            <tr
+              key={post.id}
+              data-cy="Post"
             >
-              Open
-            </button>
-          </td>
-        </tr>
+              <td data-cy="PostId">{post.id}</td>
 
-        <tr data-cy="Post">
-          <td data-cy="PostId">18</td>
+              <td data-cy="PostTitle">
+                {post.title}
+              </td>
 
-          <td data-cy="PostTitle">
-            voluptate et itaque vero tempora molestiae
-          </td>
-
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link"
-            >
-              Close
-            </button>
-          </td>
-        </tr>
-
-        <tr data-cy="Post">
-          <td data-cy="PostId">19</td>
-          <td data-cy="PostTitle">adipisci placeat illum aut reiciendis qui</td>
-
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link is-light"
-            >
-              Open
-            </button>
-          </td>
-        </tr>
-
-        <tr data-cy="Post">
-          <td data-cy="PostId">20</td>
-          <td data-cy="PostTitle">doloribus ad provident suscipit at</td>
-
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link is-light"
-            >
-              Open
-            </button>
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </div>
-);
+              <td className="has-text-right is-vcentered">
+                {selectedPost?.id === post.id
+                  ? (
+                    <button
+                      type="button"
+                      data-cy="PostButton"
+                      className="button is-link"
+                      onClick={() => {
+                        selectPost(null);
+                      }}
+                    >
+                      Close
+                    </button>
+                  )
+                  : (
+                    <button
+                      type="button"
+                      data-cy="PostButton"
+                      className="button is-link is-light"
+                      onClick={() => {
+                        selectPost(post);
+                        setIsPostSelected(!isPostSelected);
+                      }}
+                    >
+                      Open
+                    </button>
+                  )}
```

Comment:
seems more concise to customise button instead of ternary-rendering 
```suggestion
    <button
      type="button"
      data-cy="PostButton"
      className={classNames("button is-link", {
        'is-light': selectedPost?.id !== post.id
      })}
      onClick={() => {
        selectPost(null);
        if (selectedPost?.id !== post.id) {
          setIsPostSelected(!isPostSelected);
        }
      }}
    >
      {selectedPost?.id === post.id ? 'Close' : 'Open'}
    </button>
```